### ISSUE TITLE
New rule: `selector-pseudo-element-no-missing-content`

### ISSUE THREAD

**Issue by MadLittleMods** [stylelint/stylelint#482]
New rule that will complain when you forget to add a `content` declaration in your pseudo elements, `::before`, `::after`.

Name needs work: `selector-pseudo-element-no-missing-content`

It would complain about:
```css
.foo::before {
  color: #f00;
}
.bar::after {
  color: #f00;
}
```

These would be considered okay:

```css
.foo::before {
  content: 'foo';
  color: #f00;
}
.bar::after {
  content: '';
  color: #f00;
}
.qux::after {
  content: none;
  color: #f00;
}
.asdf::after {
  content: inherit;
  color: #f00;
}
```
It should work with both single and double colon syntax.

---

Someone could write `none` or `normal` but that is a use case for [`property-unit-blacklist`](https://github.com/stylelint/stylelint/blob/master/src/rules/property-unit-blacklist/README.md)

**Comment by jeddy3** [comment]
Thanks, as always, for the suggestions :)

I think we've avoided this rule up until now because, unlike the rules we've done so far, a common use case for this involves determining the cascade e.g.

```css
.ComponentName::after {
  background-color: red;
  content: "→";
}

.ComponentName-is-active::after {
  background-color: blue;
}
```

So, if we keep it simple and only apply it on a per-rule basis it'll likely behave in a manner that is at odd to the user's expectations. If we try to factor in the cascade, things are going to get complicated. 

What does everyone else think?

1. We could add it as `rule-pseudo-element-no-missing-content-property` and apply it only to the rule. We then document the limitation. Although, I'm unsure if the rule is then still useful.
2. We investigate the feasibility (both in complexity and performance) of poking around the cascade.

**Comment by MadLittleMods** [comment]
@jeddy3 I also questioned this situation for a while before I posted. I feel like this rule will probably not be developed because of the short-comings, just wanted to bring it up.

I have dealt with looking at the explicit cascade from the CSS when developing [`postcss-css-variables`](https://github.com/MadLittleMods/postcss-css-variables). It works pretty well but doesn't  support the BEM-like selectors(conscious decision). I do need to refactor that code and I am planning to split out that cascade/scope utility out into it's own module, perhaps this should be revisited then.

**Comment by jeddy3** [comment]
> I also questioned this situation for a while before I posted

I figured you would've done :)

> I do need to refactor that code and I am planning to split out that cascade/scope utility out into it's own module, perhaps this should be revisited then.

That sounds like a plan to me. But lets wait for @davidtheclark opinion before closing this issue.

**Comment by davidtheclark** [comment]
Sure, I agree. I think idea is good, but gets sullied by the fact that any given rule doesn't determine whether the selector *ever* gets `content`. If anybody finds or creates an accurate way to assess such things, let's re-open.
